upup/pkg/fi/cloudup/dns.go

```diff
@@ -170,17 +174,39 @@ func precreateDNS(cluster *api.Cluster, cloud fi.Cloud) error {
 
 	for _, dnsHostname := range dnsHostnames {
 		dnsHostname = dns.EnsureDotSuffix(dnsHostname)
-		dnsRecord := recordsMap["A::"+dnsHostname]
 		found := false
-		if dnsRecord != nil {
-			rrdatas := dnsRecord.Rrdatas()
-			if len(rrdatas) > 0 {
-				glog.V(4).Infof("Found DNS record %s => %s; won't create", dnsHostname, rrdatas)
-				found = true
-			} else {
-				// This is probably an alias target; leave it alone...
-				glog.V(4).Infof("Found DNS record %s, but no records", dnsHostname)
-				found = true
+		if cloud.ProviderID() != "vsphere" {
```

Comment:
Same comment as @SandeepPissay, let's try to collect all lose strings in form of constants.